### ISSUE TITLE
$.text() return empty string

### ISSUE THREAD

**Issue by kadamwhite** [cheeriojs/cheerio#854]
From my understanding of the docs, once you `cheerio.load( someMarkup )` then calling .html or .text on the returned variable should return the entire input markup and the input markup less HTML tags, respectively. However, `$.html()` (as used in the readme) works, but `$.text()` only returns an empty string.

Put another way, I would expect these two snippets to be equivalent, but where I would expect "Some Text String" for the result of `$1.text()`, an empty string is returned instead: `.text()` only works if a selector is provided first.

``` js
var cheerio = require('cheerio');

// comments represent actual output

console.log('Unwrapped:')
var $1 = cheerio.load('Some <em>Text</em> String');
console.log($1.html()); // "Some <em>Text</em> String"
console.log($1.text()); // ""

console.log('Wrapped:');
var $2 = cheerio.load('<h1>Some <em>Text</em> String</h1>');
console.log($2('h1').html()); // "Some <em>Text</em> String"
console.log($2('h1').text()); // "Some Text String"
```

**Comment by jugglinmike** [comment]
Hey Kadam, thanks for the report!

The trouble here is in the distinction between the static `$.text` method and
the `$.prototype.text` instance method. `$.prototype.text` has the behavior you
want, but it requires you make an instance first--a bit tricky when your
document does not have a top-level element. `$.text` would seem like the right
alternative (given the documented behavior of `$.html` as you point out), but
it doesn't actually do this... for some reason.

I've submitted gh-855 to enable your use case for `$.text` and to promote API
parity.  As I mention there, the new feature is technically backwards breaking,
although only of undocumented/untested functionality. I'm going to defer to my
fellow maintainers to decide on if/how/when we'd like to merge that patch.

In the mean time, I have two alternatives that will work for you today, both of
which use documented, tested APIs:

1. Passing the root node to the static method: `$.text($.root())`
2. Calling the instance method on a selection for the root node:
   `$.root().text()`

Do either of those work for you?

**Comment by kadamwhite** [comment]
Thank you, @jugglinmike; I have implemented a solution in line with my sample example, where I manually wrap the markup in a div with a unique ID before passing it into cheerio, and that does what I need it to do until gh-855 is decided one way or another. I appreciate the quick investigation!

**Comment by eric-burel** [comment]
Hi,

That be great to update documentation to match this tricky behaviour, it currently gives the following example :
```
$ = cheerio.load('This is <em>content</em>.')
$.text()
//=> This is content.
```
which does not seem to work.

**Comment by jackycute** [comment]
+1 here

**Comment by DivXPro** [comment]
same to me

**Comment by LuisSevillano** [comment]
+1 here

**Comment by fb55** [comment]
Fixed in #855 (not released yet)
